### DISCUSSION TITLE
Error: P1001: Can't reach database server at `"db.xxxxxxxxxxx.supabase.co:5432"`

### DISCUSSION THREAD

**Discussion by ClemSK** [supabase/supabase#7538] [Questions]
Hi all,

I'm starting a new project and going through the docs to use supabase + prisma + React but despite following the steps outlined in the quickstart I'm getting an error where I can't reach the db. I thought it might be that there was an outage of some kind, but the status seems fine - https://status.supabase.com/ .  Anyone else having this issue? 

command: **npx prisma migrate dev**      

error message:
**Environment variables loaded from prisma/.env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "postgres", schema "public" at "db.xxxxxxxxxxxxxxxxxxx.supabase.co:5432"
Error: P1001: Can't reach database server at `"db.xxxxxxxxxxxxxxxxxxx.supabase.co:5432"`
Please make sure your database server is running at `"db.xxxxxxxxxxxxxxxxxxx.supabase.co:5432"`.**

I've run through the obvious steps of checking my environment parameters and password and the connection string. 

The odd thing is that I can see that there are API calls being made to the db but am unable to access / read / write to it. Could it be a certificate issue? I've seen others mention that it might be that the postgres port has switched to 5433 but no luck there. I've also tried restarting the db in case there was an issue there, but it's unclear where my error is. 

docs: 

- https://www.prisma.io/docs/getting-started/setup-prisma/start-from-scratch/relational-databases-typescript-postgres 
- https://supabase.com/docs/guides/integrations/prisma

**Comment by burggraf** [comment]
Have you filed a support ticket regarding this?  Then our backend engineers can at least check out if there's an issue with your specific project.

**Comment by NEmbgen** [comment]
I'm having the same problem. Trying to connect from Prisma and pgAdmin4. My password does not have special characters or anything. The workaround [found here](https://github.com/supabase/supabase/discussions/8472) didn't work for me.

I have recreated the project twice, and still get the same error. This did not happen a few weeks ago.
Did something change here?

**Comment by Haack79** [comment]
I'm also having the same issue, create an entirely new project, double checked the format of my connection string and made a new password, put it in, and using it with prisma and getting the same error.

**Reply by Haack79** [reply]
never mind it was the password was too funky

**Reply by JohnQuy** [reply]
Passwords can't contain @ for this I've found as got: Error: P1001: Can't reach database server at `unnZ3yfywi:5432`

**Comment by yosegu23** [comment]
To solve this problem, the password never wrong even we never change it, I've been solved this with the [project-name] change to Reference ID not project name, you can find it from project settings in supabase dashboard, then you can find the card section general settings.

**Reply by Bryan-Herrera-DEV** [reply]
Could you explain this better? I don't have much experience with supabase

**Reply by camejiaf** [reply]
kinda late but when you go to connect (you will see the plug icon) and inside connection string you will see a link like: postgresql://postgres:[YOUR-PASSWORD]@db.XXXXXXXXXX.supabase.co:5432/postgres The X's you will be able to see them and that is you project reference ID.

**Comment by devtadanarsa** [comment]
I encountered this problem today, so I tried to restart the Supabase project. Then I realized the database URL had changed. So, I changed the env variable, and the problem was fixed.

**Reply by quang-pham-1109** [reply]
Yo this works for me. Thanks bro

**Reply by usama-sattar** [reply]
woked for me

**Reply by TheHeavyDriver** [reply]
you saved me from almost giving up on this one project :sob:, thenksss

**Comment by dyudis314** [comment]
If you're seeing this in 2025 and using Prisma, rather than using a Connection String, what worked for me:
1) Supabase > Connect > ORMs > select Prisma
2) Update your schema.prisma and ENV files accordingly
<img width="989" alt="Screenshot 2025-04-25 at 1 07 39 PM" src="https://github.com/user-attachments/assets/7c881a96-449e-4764-8edc-56221c845cd8" />

**Reply by dyudis314** [reply]
Additionally, in Vercel, make sure you add the DATABASE_URL and DIRECT_URL with their corresponding values.

**Reply by aschzczdo** [reply]
This worked for me, thank u!

**Reply by leogilmour** [reply]
thank you!

**Reply by hersariz** [reply]
it works thankyou so much

**Reply by ameetj85** [reply]
Thank you very much

**Reply by DavideOnano** [reply]
Finally a solution.
Thank you so much

**Reply by techpool** [reply]
Thank you! You the king!

**Reply by HousseinSA** [reply]
BROOOOOO, your the best thank you very Much

**Reply by Rvervuurt** [reply]
Yes, this is the solution (for now)!

**Reply by ThomasChan16** [reply]
dude idk where ur from but if you're in toronto i genuinely owe you a coffee or something. i spent HOURS AND HOURS trying EVERYTHING to figure it out. i even switch from npm to yarn and back, made new projects both on railway and supabase. WOW.

**Reply by ThomasChan16** [reply]
added you on linkedin.

**Reply by s-t-i-h-i** [reply]
thx!

**Reply by ysw753** [reply]
thx bro

**Reply by ReviDaerobi** [reply]
thanks mate, this the solution

**Reply by abdeeal** [reply]
this really helped me a lot! thanks

**Reply by tranth0ngquang** [reply]
you are GOAT! BRO SAVING LIFE !!!

**Reply by oyingrace** [reply]
> If you're seeing this in 2025 and using Prisma, rather than using a Connection String, what worked for me:
> 
> 1. Supabase > Connect > ORMs > select Prisma
> 2. Update your schema.prisma and ENV files accordingly
> 
> <img alt="Screenshot 2025-04-25 at 1 07 39 PM" width="989" src="https://private-user-images.githubusercontent.com/66389199/437541801-7c881a96-449e-4764-8edc-56221c845cd8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIyMjgzMTcsIm5iZiI6MTc1MjIyODAxNywicGF0aCI6Ii82NjM4OTE5OS80Mzc1NDE4MDEtN2M4ODFhOTYtNDQ5ZS00NzY0LThlZGMtNTYyMjFjODQ1Y2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzExVDEwMDAxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4NzEwOGE1ZWUwNmRkN2MxNTQyYWVmMmE0MzdjODIxMDdjN2FmNTdmOWNkMGRhYjE4ZTNmYmYzMWIzN2RmMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eZ-52hfY1hZpli5Nh9omPceRfpYy55OOPIXYSYhZ8eo">

If you’re still seeing this error, Go through this YouTube Video, it worked for me after trying everything else for 5 hours. 

https://youtu.be/jA2-IwR0zjk?si=x39rL6X-_TPQiN3G

**Reply by paulgasbarra** [reply]
This is the way.

**Reply by skellynb** [reply]
you are the best..im a follower now, thanks so much, ive been battling it for weeks

**Reply by Asibe-Cheta** [reply]
> > If you're seeing this in 2025 and using Prisma, rather than using a Connection String, what worked for me:
> > 
> > 1. Supabase > Connect > ORMs > select Prisma
> > 2. Update your schema.prisma and ENV files accordingly
> > 
> > <img alt="Screenshot 2025-04-25 at 1 07 39 PM" width="989" src="https://private-user-images.githubusercontent.com/66389199/437541801-7c881a96-449e-4764-8edc-56221c845cd8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIyMjgzMTcsIm5iZiI6MTc1MjIyODAxNywicGF0aCI6Ii82NjM4OTE5OS80Mzc1NDE4MDEtN2M4ODFhOTYtNDQ5ZS00NzY0LThlZGMtNTYyMjFjODQ1Y2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzExVDEwMDAxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4NzEwOGE1ZWUwNmRkN2MxNTQyYWVmMmE0MzdjODIxMDdjN2FmNTdmOWNkMGRhYjE4ZTNmYmYzMWIzN2RmMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eZ-52hfY1hZpli5Nh9omPceRfpYy55OOPIXYSYhZ8eo">
> 
> If you’re still seeing this error, Go through this YouTube Video, it worked for me after trying everything else for 5 hours.
> 
> https://youtu.be/jA2-IwR0zjk?si=x39rL6X-_TPQiN3G

i have tried this and still get the P1001 error. Thanks all the same.

**Reply by Humboorgir** [reply]
Wasn't expecting this to work. Thank you

**Reply by obedparla** [reply]
worked!

**Reply by 0xWarg2** [reply]
u saved my dayyy thank u

**Reply by sodarokr** [reply]
> > If you're seeing this in 2025 and using Prisma, rather than using a Connection String, what worked for me:
> > 
> > 1. Supabase > Connect > ORMs > select Prisma
> > 2. Update your schema.prisma and ENV files accordingly
> > 
> > <img alt="Screenshot 2025-04-25 at 1 07 39 PM" width="989" src="https://private-user-images.githubusercontent.com/66389199/437541801-7c881a96-449e-4764-8edc-56221c845cd8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTIyMjgzMTcsIm5iZiI6MTc1MjIyODAxNywicGF0aCI6Ii82NjM4OTE5OS80Mzc1NDE4MDEtN2M4ODFhOTYtNDQ5ZS00NzY0LThlZGMtNTYyMjFjODQ1Y2Q4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA3MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNzExVDEwMDAxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI4NzEwOGE1ZWUwNmRkN2MxNTQyYWVmMmE0MzdjODIxMDdjN2FmNTdmOWNkMGRhYjE4ZTNmYmYzMWIzN2RmMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eZ-52hfY1hZpli5Nh9omPceRfpYy55OOPIXYSYhZ8eo">
> 
> If you’re still seeing this error, Go through this YouTube Video, it worked for me after trying everything else for 5 hours.
> 
> https://youtu.be/jA2-IwR0zjk?si=x39rL6X-_TPQiN3G

THIS SAVED MY WHOLE DAY!

**Reply by luciemtro** [reply]
Thank you !!!

**Reply by HuseinHQ** [reply]
I’m using Prisma and it helped. Thanks!

**Reply by kalabhaftu** [reply]
thanks this worked for me :)

**Reply by PURPLE-ORCA** [reply]
This solved my problem, Thanks

**Reply by juceledonc** [reply]
thanks buddy

**Reply by Dev-Dayo20** [reply]
Thanks buddy, this solved my problem, being battling with it for hours

**Reply by daviddprtma** [reply]
wow thank you so much for this help. you save my hours🚀

**Reply by anshahkhan** [reply]
It solved the problem, Thanks bro <3

**Reply by Ceir-Ceir** [reply]
worked for me thanks!

**Reply by SasaniP** [reply]
Worked for me. Thanks a lot!

**Comment by engalisabry** [comment]
## How I Solved Prisma Migration Error with Supabase (P1001)

After a lot of trial and error — deleting and recreating the Supabase project, changing the database password, and attempting migrations multiple times — I finally found a solution.

In my case, I was using **Prisma with Supabase**.

- Testing the database connection manually using `psql` **worked fine**.
- Running a Prisma migration **kept failing** with a connection error (P1001).

---

###  Solution

I added **timeout parameters** to both `DATABASE_URL` and `DIRECT_URL` in my `.env` file.

---

### ❌ Old Version (was failing)

```env
# Pooled connection for Prisma Client
DATABASE_URL="postgresql://postgres:[YOUR_PASSWORD]@aws-0-us-east-2.pooler.supabase.com:5432/postgres?pgbouncer=true"

# Direct connection for Prisma Migrate
DIRECT_URL="postgresql://postgres:[YOUR_PASSWORD]@db.[PROJECT_REF].supabase.co:5432/postgres"

```
### ✅ New Version (worked for me)


```env
# Pooled connection for Prisma Client
DATABASE_URL="postgresql://postgres:[YOUR_PASSWORD]@aws-0-us-east-2.pooler.supabase.com:5432/postgres?pgbouncer=true&connect_timeout=30&pool_timeout=30"

# Direct connection for Prisma Migrate
DIRECT_URL="postgresql://postgres:[YOUR_PASSWORD]@db.[PROJECT_REF].supabase.co:5432/postgres?connect_timeout=30"

```

Hope this helps someone else struggling with the same issue! 🚀

**Reply by LordKobi** [reply]
This worked for me, Bless You)

**Reply by oyingrace** [reply]
You’re welcome

**Reply by Gray-Mafutala** [reply]
Thanks a lot bro !!!

**Comment by Jasen2009** [comment]
i am getting this error everytime: 

prisma:warn Prisma failed to detect the libssl/openssl version to use, and may not work as expected. Defaulting to "openssl-1.1.x".
Please manually install OpenSSL and try installing Prisma again.
prisma:warn Prisma failed to detect the libssl/openssl version to use, and may not work as expected. Defaulting to "openssl-1.1.x".
Please manually install OpenSSL and try installing Prisma again.
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma

Datasource "db": PostgreSQL database "postgres", schema "public" at "db.eyeedgpkhpxdvdrlqqws.supabase.co:5432"
Error: Schema engine error:

**Comment by marwalabhi** [comment]
Go to connect tab and in this find ORMs, and in this section copy the code for prisma and paste in your workspace, it definitely works 🙏🙌

**Comment by LunaT93** [comment]
The right answer was here but I didn't read it because of someone's flashy (but probably incidentally correct for them) post. **_Go to the ORMs section of Supabase and copy THOSE_**. I tried everything under the sun with redwood's documentation (the framework I used), netlify docs, substack, asking copilot (it's normally decent but was totally unhelpful about this issue). Anyway, you still need real humans toiling for two days and duplicating the answer that @marwalabhi gave 3 weeks ago. Go to Dashboard --> Connect --> ORMs. Put those wherever your .env values need to go (Netlify's environmental variables section for me).

**Comment by nasirhussainn** [comment]
I ran into the same P1001 error when trying to connect Prisma with Supabase on the free plan.
The issue was that the Direct Connection uses IPv6 only, while my network is IPv4.

Switch to the Session Pooler connection string, which is IPv4-compatible. Example:

`postgresql://postgres.xxxxxxxxxxx:[YOUR-PASSWORD]@aws-1-ap-southeast-1.pooler.supabase.com:5432/postgres?sslmode=require`


Also make sure your password is URL-encoded if it contains special characters like @, #, or ! — otherwise Prisma may fail to parse the connection string properly. You can encode it easily using a site like https://www.urlencoder.org/


After switching to the session pooler and encoding the password, Prisma connected successfully

**Comment by Shubham0523** [comment]
It’s an IP-related issue. Make sure IPv6 is enabled on both your router and your device.

**Comment by Davda-James** [comment]
It works but really its too irritating that I have to restart database each time to make it work whenever I face this error and I am facing it at least more than four times in literally 3 to 4 days
